src/overlays/actors/ovl_En_Elf/z_en_elf.c

```diff
@@ -21,99 +66,1503 @@ const ActorInit En_Elf_InitVars = {
+    static Vec3f sparkleVelocity = { 0.0f, -0.05f, 0.0f };
+    static Vec3f sparkleAccel = { 0.0f, -0.025f, 0.0f };
+    s32 pad;
+    Vec3f sparklePos;
+    Color_RGBA8 primColor;
+    Color_RGBA8 envColor;
+
+    sparklePos.x = Math_Rand_CenteredFloat(6.0f) + this->actor.posRot.pos.x;
+    sparklePos.y = (Math_Rand_ZeroOne() * 6.0f) + this->actor.posRot.pos.y;
+    sparklePos.z = Math_Rand_CenteredFloat(6.0f) + this->actor.posRot.pos.z;
+
+    primColor.r = this->innerColor.r;
+    primColor.g = this->innerColor.g;
+    primColor.b = this->innerColor.b;
+
+    envColor.r = this->outerColor.r;
+    envColor.g = this->outerColor.g;
+    envColor.b = this->outerColor.b;
+
+    EffectSsKiraKira_SpawnDispersed(globalCtx, &sparklePos, &sparkleVelocity, &sparkleAccel, &primColor, &envColor,
+                                    1000, sparkleLife);
+}
+
+void func_80A04D90(EnElf* this, GlobalContext* globalCtx) {
+    s32 pad;
+    s32 bgId;
+
+    this->actor.groundY = func_8003CA0C(globalCtx, &globalCtx->colCtx, &this->actor.floorPoly, &bgId, &this->actor,
+                                        &this->actor.posRot.pos);
+    this->actor.shape.unk_14 = 0x32;
+}
+
+// move to talk to player
+void func_80A04DE4(EnElf* this, GlobalContext* globalCtx) {
+    Vec3f posRot2Copy;
+    Player* player = PLAYER;
+    Vec3f naviRefPos;
+
+    if (this->fairyFlags & 0x10) {
+        naviRefPos = globalCtx->actorCtx.targetCtx.naviRefPos;
+
+        if (((player->unk_664 == NULL) || (&player->actor == player->unk_664)) || (&this->actor == player->unk_664)) {
+            naviRefPos.x = player->bodyPartsPos[7].x + (Math_Sins(player->actor.shape.rot.y) * 20.0f);
+            naviRefPos.y = player->bodyPartsPos[7].y + 5.0f;
+            naviRefPos.z = player->bodyPartsPos[7].z + (Math_Coss(player->actor.shape.rot.y) * 20.0f);
+        }
+
+        this->actor.posRot2.pos = naviRefPos;
+        this->fairyFlags &= ~0x10;
+    }
+
+    func_80A03AB0(this, globalCtx);
+    posRot2Copy = this->actor.posRot2.pos;
+
+    func_80A03148(this, &posRot2Copy, 0, 20.0f, 0.2f);
+
+    if (this->actor.speedXZ >= 5.0f) {
+        EnElf_SpawnSparkles(this, globalCtx, 16);
+    }
+
+    Math_SmoothScaleMaxMinF(&this->actor.scale.x, 0.008f, 0.3f, 0.00080000004f, 0.000080000005f);
+    EnElf_UpdateLights(this, globalCtx);
+}
+
+// move after talking to player
+void func_80A04F94(EnElf* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+
+    Math_SmoothScaleMaxMinS(&this->actor.shape.rot.y, this->unk_2BC, 5, 0x1000, 0x400);
+    this->timer++;
+    Math_ApproxF(&this->unk_2A4, 1.0f, 0.05f);
+    func_800773A8(globalCtx, SQ(this->unk_2A4), player->actor.projectedPos.z + 780.0f, 0.2f, 0.5f);
+}
+
+// ask to talk to saria again
+void func_80A05040(Actor* thisx, GlobalContext* globalCtx) {
+    EnElf* this = THIS;
+
+    func_80A04DE4(this, globalCtx);
+
+    if ((func_8010BDBC(&globalCtx->msgCtx) == 4) && (func_80106BC8(globalCtx))) {
```

Comment:
```suggestion
    if ((func_8010BDBC(&globalCtx->msgCtx) == 4) && func_80106BC8(globalCtx)) {
```
there's another `&& (func_80106BC8(globalCtx))` in the file